model_compression_toolkit/core/common/graph/virtual_activation_weights_node.py

```diff
@@ -0,0 +1,198 @@
+            weights: Dictionary from a variable name to the weights with that name in the layer the node represents.
+            layer_class: Class path of the layer this node represents.
+            reuse: Whether this node was duplicated and represents a reused layer.
+            reuse_group: Name of group of nodes from the same reused layer.
+            quantization_attr: Attributes the node holds regarding how it should be quantized.
+            has_activation: Whether the node has activations that we might want to quantize.
+            **kwargs: Additional arguments that can be passed but are not used (allows to init the object with an
+                existing node's __dict__).
+
+        """
+
+        super().__init__(name,
+                         framework_attr,
+                         input_shape,
+                         output_shape,
+                         weights,
+                         layer_class,
+                         reuse,
+                         reuse_group,
+                         quantization_attr,
+                         has_activation)
+
+        self.name = f"{VIRTUAL_ACTIVATION_WEIGHTS_NODE_PREFIX}_{act_node.name}_{weights_node.name}"
+
+        self.original_activation_node = act_node
+        self.original_weights_node = weights_node
+
+        v_candidates = []
+        for c_a in act_node.candidates_quantization_cfg:
+            for c_w in weights_node.candidates_quantization_cfg:
+                composed_candidate = CandidateNodeQuantizationConfig(activation_quantization_cfg=c_a.activation_quantization_cfg,
+                                                                     weights_quantization_cfg=c_w.weights_quantization_cfg)
+                v_candidates.append(composed_candidate)
+
+        # sorting the candidates by weights number of bits first and then by activation number of bits (reversed order)
+        v_candidates.sort(key=lambda c: (c.weights_quantization_cfg.weights_n_bits,
+                                         c.activation_quantization_cfg.activation_n_bits), reverse=True)
+
+        self.candidates_quantization_cfg = v_candidates
+
+    def get_bops_count(self, fw_impl: Any, fw_info: FrameworkInfo, candidate_idx: int) -> float:
+        """
+        Computes the composed node's (edge) bit-operation count.
+
+        Args:
+            fw_impl: A FrameworkImplementation object with framework specific methods.
+            fw_info: A FrameworkInfo object with framework specific information,
+            candidate_idx: The index of the node's quantization candidate configuration.
+
+        Returns: The BOPS count of the composed node.
+
+        """
+        node_mac = fw_impl.get_node_mac_operations(self.original_weights_node, fw_info)
+        node_bops = self.candidates_quantization_cfg[candidate_idx].weights_quantization_cfg.weights_n_bits * \
+                    self.candidates_quantization_cfg[candidate_idx].activation_quantization_cfg.activation_n_bits * \
```

Comment:
I'm guessing you did it this way cause the activation node is preceding the weights node, right?
If so - please add a comment for it. Otherwise, I think it's a bug

Reply:
Yes, this is the whole point of this structure (the VirtualActivationWeightsNode class). This is not the activation of the Convolution that the node holds but the activation that precedes it. I'll explain this in the documentation.